### PR TITLE
Dmp 58 setup app insights

### PR DESCRIPTION
**Pull request by alancoleman**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-58

### Change description ###
Attempt to service logs to App Insights

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: infrastructure/.terraform-version [hunk 1]
```diff
@@ -1 +0,0 @@
-1.4.4
```
Comment (jackmaloney) [inline infrastructure/.terraform-version:1]: Did we want to be removing this?

File: lib/applicationinsights.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+  "connectionString": "${file:/mnt/secrets/draft-store/app-insights-connection-string}",
   "role": {
     "name": "rpe-demo"
```
Comment (jackmaloney) [inline lib/applicationinsights.json:4]: This role name needs to be updated to something DARTS related I think

File: infrastructure/variables.tf [hunk 3]
```diff
@@ -1,15 +1,18 @@
-variable "product" {}
-
-variable "component" {}
-
-variable "location" {
-  default = "UK South"
+variable "resource_group_name" {
```
Comment (jackmaloney) [inline infrastructure/variables.tf:1]: Did you mean to remove all of the other fields that existed here?

**Review by jackmaloney — COMMENTED** [review]
Did we not want to follow the pattern defined in this file? Or is there a reason that's not compatible? https://github.com/hmcts/draft-store/blob/master/infrastructure/app-insights.tf

File: infrastructure/.terraform-version (hunk 1 above)
Comment (alancoleman) [reply infrastructure/.terraform-version:1]: Thanks @jackmaloney, the Terraform version is described in providers.tf and the plan executes on Demo without this file, however it's probably a little naive of me to remove it so I'll add it back in.

Comment (alancoleman) [reply infrastructure/.terraform-version:1]: I did notice that removing this file create a conflict anyway. So adding it back in makes sense.

File: lib/applicationinsights.json (hunk 2 above)
Comment (alancoleman) [reply lib/applicationinsights.json:4]: I think I'll need to add the required Azure resources before attempting to connect or pass anything to them. That's why I left the application part for another commit. Thanks

File: infrastructure/variables.tf (hunk 3 above)
Comment (alancoleman) [reply infrastructure/variables.tf:1]: Thanks @jackmaloney,  yes, location, name and tags are the only arguments available when creating an Azure Resource Group, didn't want anything else confusing matters. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_gro
up

**Comment by alancoleman** [comment]
> Did we not want to follow the pattern defined in this file? Or is there a reason that's not compatible? https://github.com/hmcts/draft-store/blob/master/infrastructure/app-insights.tf

Thanks @jackmaloney, it looks as though most of the draft-store infrastructure directory is applicable to the draft-store app, the database set up etc, not just the app insights. I think what we're looking for is a lot simpler: Resource Group, App Insights, Azure Key and then connect the app with the resulting strings. Kicking things off with a Resource Group will at least give me an intro into how this process works and what basic config is required. Happy to discuss. Thanks

**Pull request closed without merging**